lib/screens/recording_screen.dart

```diff
@@ -0,0 +1,274 @@
+import 'dart:math' show pi;
+import 'dart:io' show File, Directory;
+import 'dart:async' show StreamSubscription;
+
+import 'package:flutter/material.dart';
+import 'package:wave/wave.dart';
+import 'package:wave/config.dart';
+
+import 'package:intl/intl.dart' show DateFormat;
+import 'package:flutter_sound/flutter_sound.dart';
+
+import '../models/sentence_model.dart';
+
+class RecordingScreen extends StatefulWidget {
+  final Sentence phrase;
+  final String targetLang = "Yemba";
+
+  RecordingScreen({Key key, @required this.phrase})
+      : assert(phrase != null),
+        super(key: key);
+
+  @override
+  _RecordingScreenState createState() => _RecordingScreenState();
+}
+
+class _RecordingScreenState extends State<RecordingScreen> {
+  String _recorderTimer;
+  bool _isRecording;
+  double _recorderIconSize = 60.0;
+  FlutterSound _flutterSound;
+  String _recordStorage = "/sdcard/Lingua";
+  StreamSubscription _recorderSubscription;
+
+  void _startRecording() {
+    String fileId = widget.phrase.id;
+    new File("$_recordStorage/$fileId.m4a").create().then((filePath) {
+      _flutterSound.startRecorder(filePath.path).then((path) {
+        setState(() {
+          this._isRecording = true;
+        });
+
+        print('startRecorder, file is: $path');
+
+        _recorderSubscription =
+            _flutterSound.onRecorderStateChanged.listen((e) {
+          if (e == null) {
+            return;
+          }
+          DateTime date = new DateTime.fromMillisecondsSinceEpoch(
+              e.currentPosition.toInt());
+          String txt = DateFormat('mm:ss:SS', 'en_US').format(date);
+
+          setState(() {
+            this._recorderTimer = txt.substring(0, 8);
+          });
+        });
+      }).catchError((error) {
+        print(error);
+      });
+    }); //File
+  }
+
+  void _stopRecording() {
+    _flutterSound.stopRecorder().then((value) {
+      print('stopRecorder: $value');
+      if (_recorderSubscription != null) {
+        _recorderSubscription.cancel();
+        _recorderSubscription = null;
+      }
+
+      setState(() {
+        this._isRecording = false;
+      });
+    }).catchError((error) {
+      print("Was not able to stop recorder: $error");
```

Comment:
This should probably be some user scree error? I think we should avoid printing in production code.